Orange/widgets/visualize/owpythagorastree.py

```diff
@@ -0,0 +1,416 @@
+        t0 = self._get_point_on_square_edge(
+            parent_center, parent_length, parent_angle)
+        # get the edge point that we will rotate around t0
+        square_diagonal_length = math.sqrt(2 * parent_length ** 2)
+        edge = self._get_point_on_square_edge(
+            parent_center, square_diagonal_length, parent_angle - PI / 4)
+        # if the new square is not the first child, we need to rotate the edge
+        if base_angle != 0:
+            edge = self._rotate_point(edge, t0, base_angle)
+
+        # rotate the edge point to the correct spot
+        t1 = self._rotate_point(edge, t0, alpha)
+
+        # calculate the middle point between the rotated point and edge
+        t2 = Point((t1.x + edge.x) / 2, (t1.y + edge.y) / 2)
+        # calculate the slope of the new square
+        slope = parent_angle - PI / 2 + alpha / 2
+        # using this data, we can compute the square center
+        return self._get_point_on_square_edge(t2, length, slope + base_angle)
+
+    @staticmethod
+    def _rotate_point(point, around, alpha):
+        """Rotate a point around another point by some angle.
+
+        Parameters
+        ----------
+        point : Point
+            The point to rotate.
+        around : Point
+            The point to perform rotation around.
+        alpha : float
+            The angle to rotate by (in radians).
+
+        Returns
+        -------
+        Point:
+            The rotated point.
+
+        """
+        temp = Point(point.x - around.x, point.y - around.y)
+        temp = Point(
+            temp.x * math.cos(alpha) - temp.y * math.sin(alpha),
+            temp.x * math.sin(alpha) + temp.y * math.cos(alpha)
+        )
+        return Point(temp.x + around.x, temp.y + around.y)
+
+    @staticmethod
+    def _get_point_on_square_edge(center, length, angle):
+        """Calculate the central point on the drawing edge of the given square.
+
+        Parameters
+        ----------
+        center : Point
+            The square center point.
+        length : float
+            The square side length.
+        angle : float
+            The angle of the square.
+
+        Returns
+        -------
+        Point
+            A point on the center of the drawing edge of the given square.
+
+        """
+        return Point(
+            center.x + length / 2 * math.cos(angle),
+            center.y + length / 2 * math.sin(angle)
+        )
+
+
+class SklTreeAdapter:
+    def __init__(self, tree):
+        self._tree = tree
+
+    def weight(self, node):
+        return self.num_samples(node) / self.num_samples(self.parent(node))
+
+    def num_samples(self, node):
+        return self._tree.n_node_samples[node]
+
+    def parent(self, node):
+        for idx, el in enumerate(self._tree.children_left):
+            if el == node:
+                return idx
+        for idx, el in enumerate(self._tree.children_right):
+            if el == node:
+                return idx
+        return -1
```

Comment:
If you say this is slow, and the `_tree` is from sklearn; can you test if this works and is faster?
```py
for children in (self._tree.children_left,
                 self._tree.children_right):
    try:
        return (children == node).nonzero()[0][0]
    except IndexError:
        continue
return -1
```
It lays the iterative burden onto numpy.